jane/doc/parallelism-tutorial.md

```diff
@@ -77,36 +197,71 @@ let%expect_test "add4 in parallel" =
 This creates a work-stealing scheduler, along with a _monitor_ to manage
 exceptions. Then it tells the scheduler to run the `test_add4` function before
 shutting down the scheduler. (Naturally, a real program will want to keep the
-monitor and scheduler around longer!) To test using the sequential scheduler
-instead, we would simply replace `Parallel_scheduler_work_stealing` with
-`Parallel.Scheduler.Sequential`.
+monitor and scheduler around longer!)
+
+You can also use the `parallel` library's own `Parallel.Scheduler.Sequential`,
+which simply runs everything on the primary domain. This is handy for testing or
+debugging when you want to eliminate nondeterminism. To do so, simply replace
+`Parallel_scheduler_work_stealing` with `Parallel.Scheduler.Sequential`.
+
+Now for something more substantial. Suppose we're working with binary trees, and
+we want to take an average over all the values in the tree. Here's a basic
+implementation:
+
+<a id="code-average"></a>
+```ocaml
+module Tree = struct
+  type 'a t =
+    | Leaf of 'a
+    | Node of 'a t * 'a t
+end
+
+let average (tree : float Tree.t) =
+  let rec total tree : total:float * count:int =
+    match tree with
+    | Tree.Leaf x -> ~total:x, ~count:1
+    | Tree.Node (l, r) ->
+      let ~total:total_l, ~count:count_l = total l in
+      let ~total:total_r, ~count:count_r = total r in
+      ~total:(total_l +. total_r), ~count:(count_l + count_r)
+  in
+  let ~total, ~count = total tree in
+  total /. (count |> Float.of_int)
+;;
+```
+
+(Note that we've made use of the new [labeled tuples] feature for the values
+returned by `total`.)
```

Comment:
I would move this note above the code. Otherwise, readers' OCaml parsers will fail and they may not make it here.